### PR TITLE
Add Blake 1.3.3

### PR DESCRIPTION
**Pull request by IanBullard**
Add blake 1.3.3

### CODE REVIEW

File: packages/b/blake3/xmake.lua [hunk 1]
```diff
@@ -28,6 +29,12 @@ package("blake3")
                     else
                         add_files("c/*x86-64_unix.S")
                     end
+                elseif is_arch("x86") then
```
Comment (waruqi) [inline packages/b/blake3/xmake.lua:32]: `is_arch("x86", "i386")`

**Comment by waruqi** [comment]
error: @programdir\modules\private\async\runjobs.lua:256: @programdir\modules\private\action\build\object.lua:91: @programdir\modules\core\tools\gcc.lua:709: D:/a/_temp/msys64/mingw32/lib/gcc/i686-w64-mingw32/12.2.0/include/avxintrin.h:927:1: error: inlining failed in call to 'always_inline' '_mm256_loadu_si256': target specific option mismatch
  927 | _mm256_loadu_si256 (__m256i_u const *__P)
      | ^~~~~~~~~~~~~~~~~~
c\blake3_avx2.c:8:10: note: called from here
    8 |   return _mm256_loadu_si256((const __m256i *)src);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  > in c\blake3_avx2.c

maybe it does not support mingw/i386?

you can disable it `on_install("mingw|x86_64")`

File: packages/b/blake3/xmake.lua [hunk 2]
```diff
@@ -28,6 +33,12 @@ package("blake3")
                     else
                         add_files("c/*x86-64_unix.S")
                     end
+                elseif is_arch("x86", "i386", "i686") then
```
Comment (waruqi) [inline packages/b/blake3/xmake.lua:36]: we need not i686, it is also i386 for mingw platform.

Comment (IanBullard) [reply packages/b/blake3/xmake.lua:36]: Right, I wasn't sure, I'll correct this with my next commit.

**Comment by IanBullard** [comment]
I have to stop for tonight, even if the checks complete, please do not merge.

**Comment by IanBullard** [comment]
@waruqi How do I disable MingW (Msys2) static/shared i686?  Those are the only two failed builds and it's become obvious they're not supported.

**Comment by waruqi** [comment]
> @waruqi How do I disable MingW (Msys2) static/shared i686? Those are the only two failed builds and it's become obvious they're not supported.

> you can disable it on_install("mingw|x86_64")

File: packages/b/blake3/xmake.lua [hunk 3]
```diff
@@ -8,11 +8,11 @@ package("blake3")
     add_versions("1.3.3", "27d2bc4ee5945ba75434859521042c949463ee7514ff17aaef328e23ef83fec0")
     add_versions("1.3.1", "112becf0983b5c83efff07f20b458f2dbcdbd768fd46502e7ddd831b83550109")
 
-    on_install("msys2|i686")
+    on_install("mingw|x86_64")
```
Comment (waruqi) [inline packages/b/blake3/xmake.lua:11]: no

```
    on_install("windows", "linux", "macosx", "bsd", "mingw|x86_64", "android", "iphoneos", "cross", function (package)
```

File: packages/b/blake3/xmake.lua [hunk 4]
```diff
@@ -8,11 +8,11 @@ package("blake3")
...
 
     on_install(function (package)
```
Comment (waruqi) [inline packages/b/blake3/xmake.lua:13]: you need modify this

File: packages/b/blake3/xmake.lua (hunk 3 above)
Comment (IanBullard) [reply packages/b/blake3/xmake.lua]: Thank you.

**Pull request merged**